tensorboard/plugins/base_plugin.py

```diff
@@ -32,16 +32,21 @@ class TBPlugin(object):
   """TensorBoard plugin interface. Every plugin must extend from this class.
 
   Subclasses must have a constructor that takes a TBContext argument.
+
+  Fields:
+    plugin_name: The plugin_name will also be a prefix in the http handlers 
+        generated by the plugin, e.g. `data/plugins/$PLUGIN_NAME/$HANDLER` The 
+        plugin name must be unique for each registered plugin, or a ValueError 
+        will be thrown when the application is constructed. The plugin name must 
+        only contain characters among [A-Za-z0-9_.-], and must be nonempty, 
+        or a ValueError will similarly be thrown.
+    disable_path_prefix: If True then data/plugins/$PLUGIN_NAME prefix will not
```

Comment:
Hmm…I don't think that I like allowing arbitrary plugins to do this. There shouldn't be any reason that new plugins _need_ to do this, so it just adds another piece of bloat to the API—and if a plugin _were_ to do this it would just cause confusion and distinguish it unnecessarily from normal plugins.

What do you think about either replacing this field with a hard-coded `type(plugin) == CorePlugin` check in the `application` code?

I would also support just changing the client-side logic to grab things from `/data/plugin/core/logdir` and `/data/plugin/core/runs`. [^1] [^2] This leaves the issue of `index.html` and whatever static assets might be loaded from the zipdir. I wonder whether these really should be part of a plugin? (Cf. my top-level comment.)

[^1]: As far as I'm concerned, we can drop the `/graphs`/… family of routes.
[^2]. For what it's worth, I prefer a `type(−)` check over a `−.plugin_name == "−"` check. Out-of-band > in-band.